.github/workflows/ci.yml

```diff
@@ -30,11 +30,20 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
-          path: custom
+          path: custom 
+
+      - name: Install Python dependencies for custom project generation
+        run: python -m pip install pyyaml
+      
+      - name: Test python scripts
+        run: |
```

Comment:
You don't need to cd back as steps are independent.

You can even simplify with

```yaml
    name: xxx
    working-directory: custom
    env:
      PYTHONPATH: src/
    run: python -m unittest
```

The PYTHONPATH is so you don't fumble with paths in source code

Reply:
Thank you, It's good to know. I've updated it.